src/main/kotlin/lotto/domain/LottoNumber.kt

```diff
@@ -0,0 +1,30 @@
+package lotto.domain
+
+class LottoNumber private constructor(private val number: Int) {
+    fun getNumber() = number
+
+    companion object {
+        const val INVALID_MESSAGE = "은/는 로또 번호가 아닙니다. 1 ~ 45사이의 값을 넣어주세요."
+        const val MINIMUM_NUMBER = 1
+        const val MAXIMUM_NUMBER = 45
+        private val NUMBERS: Map<Int, LottoNumber> = (MINIMUM_NUMBER..MAXIMUM_NUMBER).associateWith(::LottoNumber)
+
+        fun of(value: Int): LottoNumber {
+            return NUMBERS.getOrElse(value) { throw IllegalArgumentException("$value$INVALID_MESSAGE") }
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
            return NUMBERS[value] ?: throw IllegalArgumentException("$value$INVALID_MESSAGE")
```